Build failure: nanopb, nanopbMalloc

### Steps To Reproduce

Steps to reproduce the behavior:
1. build *nanopb*, *nanopbMalloc*

### Build log

> `nanopb`
```
error: builder for '/nix/store/k8fjaylrznw9d9386i5b5vv4k370rd85-nanopb-generator-out-0.4.8.drv' failed with exit code 2;
       last 10 log lines:
       > 
       > -- Build files have been written to: /build/source/build
       > cmake: enabled parallel building
       > cmake: enabled parallel installing
       > Running phase: buildPhase
       > build flags: -j16 SHELL=/nix/store/4bj2kxdm1462fzcc2i2s4dn33g2angcc-bash-5.2p32/bin/bash
       > /nix/store/4bj2kxdm1462fzcc2i2s4dn33g2angcc-bash-5.2p32/bin/bash: line 1: /build/source/generator/protoc: cannot execute: required file not found
       > make[2]: *** [CMakeFiles/nanopb_generator.dir/build.make:74: CMakeFiles/nanopb_generator] Error 127
       > make[1]: *** [CMakeFiles/Makefile2:83: CMakeFiles/nanopb_generator.dir/all] Error 2
       > make: *** [Makefile:136: all] Error 2
       For full logs, run 'nix log /nix/store/k8fjaylrznw9d9386i5b5vv4k370rd85-nanopb-generator-out-0.4.8.drv'.
error: 1 dependencies of derivation '/nix/store/54g6pfys9ipj2x1h392rxjvhmwmmbpq0-nanopb-generator-0.4.8.drv' failed to build
error: 1 dependencies of derivation '/nix/store/rxvf1vli2x51i66wfjhk6g2kjv7d0zj8-nanopb-0.4.8.drv' failed to build
```

> `nanopbMalloc`

```
error: builder for '/nix/store/k8fjaylrznw9d9386i5b5vv4k370rd85-nanopb-generator-out-0.4.8.drv' failed with exit code 2;
       last 10 log lines:
       > 
       > -- Build files have been written to: /build/source/build
       > cmake: enabled parallel building
       > cmake: enabled parallel installing
       > Running phase: buildPhase
       > build flags: -j16 SHELL=/nix/store/4bj2kxdm1462fzcc2i2s4dn33g2angcc-bash-5.2p32/bin/bash
       > /nix/store/4bj2kxdm1462fzcc2i2s4dn33g2angcc-bash-5.2p32/bin/bash: line 1: /build/source/generator/protoc: cannot execute: required file not found
       > make[2]: *** [CMakeFiles/nanopb_generator.dir/build.make:74: CMakeFiles/nanopb_generator] Error 127
       > make[1]: *** [CMakeFiles/Makefile2:83: CMakeFiles/nanopb_generator.dir/all] Error 2
       > make: *** [Makefile:136: all] Error 2
       For full logs, run 'nix log /nix/store/k8fjaylrznw9d9386i5b5vv4k370rd85-nanopb-generator-out-0.4.8.drv'.
error: 1 dependencies of derivation '/nix/store/54g6pfys9ipj2x1h392rxjvhmwmmbpq0-nanopb-generator-0.4.8.drv' failed to build
error: 1 dependencies of derivation '/nix/store/mdh1ckpzmpa9k4x6xgqz9qx87zvjsbh4-nanopb-0.4.8.drv' failed to build
```

### Additional context

Add any other context about the problem here.

### Notify maintainers

<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

@kalbasit  @liarokapisv 

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.36, NixOS, 24.11 (Vicuna), 24.11.20240703.9f4128e`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.4`
 - channels(root): `"nixos-23.05"`
 - channels(sigmanificient): `""`
 - nixpkgs: `/nix/store/dk2rpyb6ndvfbf19bkb2plcz5y3k8i5v-source`
```
---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/na/nanopb/generator-out.nix
+++ b/pkgs/by-name/na/nanopb/generator-out.nix
@@ -1,11 +1,13 @@
-{ stdenv
-, cmake
-, python3
-, writeTextFile
-, protobuf
-, src
-, version
+{
+  stdenv,
+  cmake,
+  python3,
+  writeTextFile,
+  protobuf,
+  src,
+  version,
 }:
+
 let
   pyproject_toml = writeTextFile {
     name = "pyproject.toml";
@@ -38,22 +40,26 @@ stdenv.mkDerivation {
   pname = "nanopb-generator-out";
   inherit src version;
 
-  nativeBuildInputs = [ cmake protobuf python3 ];
+  nativeBuildInputs = [
+    cmake
+    protobuf
+  ];
+
+  # don't let `find_program` find the bundled `protoc` script, so it will use the system `protoc` instead
+  preConfigure = ''
+    rm generator/protoc
+  '';
 
   cmakeFlags = [
     "-Dnanopb_BUILD_RUNTIME=OFF"
     "-Dnanopb_BUILD_GENERATOR=ON"
-    "-Dnanopb_PYTHON_INSTDIR_OVERRIDE=$out/lib/python/site-packages"
+    "-Dnanopb_PYTHON_INSTDIR_OVERRIDE=${placeholder "out"}/${python3.sitePackages}"
   ];
-
-  preConfigure = ''
-    cmakeFlags+=" -Dnanopb_PYTHON_INSTDIR_OVERRIDE=$out/lib/python/site-packages"
-  '';
 
   postInstall = ''
     rm -rf $out/include
     rm -rf $out/lib/cmake
-    ln -s $out/lib/python/site-packages $out/src
+    ln -s $out/${python3.sitePackages} $out/src
     ln -s ${pyproject_toml} $out/pyproject.toml
   '';
 }

